pmd-test/src/main/java/net/sourceforge/pmd/lang/ParserOptionsTestUtils.java

```diff
@@ -5,38 +5,9 @@
 package net.sourceforge.pmd.lang;
 
 import org.junit.Assert;
-import org.junit.Test;
 
-/**
- * Unit tests for {@link ParserOptions}.
- */
-public class ParserOptionsTest {
-
-    /**
-     * SuppressMarker should be initially null and changeable.
-     */
-    @Test
-    public void testSuppressMarker() {
-        ParserOptions parserOptions = new ParserOptions();
-        Assert.assertNull(parserOptions.getSuppressMarker());
-        parserOptions.setSuppressMarker("foo");
-        Assert.assertEquals("foo", parserOptions.getSuppressMarker());
-    }
-
-    /**
-     * Verify that the equals and hashCode methods work as expected.
-     */
-    @Test
-    public void testEqualsHashcode() {
-        ParserOptions options1 = new ParserOptions();
-        options1.setSuppressMarker("foo");
-        ParserOptions options2 = new ParserOptions();
-        options2.setSuppressMarker("bar");
-        ParserOptions options3 = new ParserOptions();
-        options3.setSuppressMarker("foo");
-        ParserOptions options4 = new ParserOptions();
-        options4.setSuppressMarker("bar");
-        verifyOptionsEqualsHashcode(options1, options2, options3, options4);
+public final class ParserOptionsTestUtils {
```

Comment:
The renaming and the new constructor are binary-incompatible changes. To preserve binary compatibility we can declare this ParserOptionTestUtils next to ParserOptionsTest, give it a package-private constructor, and make ParserOptionsTest extend from it. And deprecate ParserOptionsTest.

Reply:
I have reverted the original file and marked it as deprecated. I renamed the file in src/test to `ParserOptionsUnitTest` with a comment to change the name in 7.0. The common piece of code is a static method, having `ParserOptionsTest` extend the util with the static method would not have retained compatibility with anyone code that is calling the ParserOptionsTest static method.